app/src/org/commcare/activities/UpdateActivity.java

```diff
@@ -325,26 +328,36 @@ public void handleTaskCompletion(ResultAndError<AppInstallStatus> result) {
         uiController.refreshView();
     }
 
-    private void reportFailureToNotifications(String errorMessage) {
+    private void reportFailureToNotifications(AppInstallStatus status, String errorMessage) {
         NotificationMessage notificationMessage = null;
 
-        if (UpdateHelper.isCombinedErrorMessage(errorMessage)) {
-            Pair<String, String> resourceAndMessage =
-                    UpdateHelper.splitCombinedErrorMessage(errorMessage);
-            notificationMessage =
-                    NotificationMessageFactory.message(AppInstallStatus.InvalidResource,
-                            new String[]{null, resourceAndMessage.first, resourceAndMessage.second});
-        } else if (!"".equals(errorMessage)) {
-            notificationMessage =
-                    NotificationMessageFactory.message(AppInstallStatus.UpdateFailedGeneral,
-                            new String[]{null, errorMessage, null});
-        }
+        switch (status) {
+            case BadSslCertificate:
+                notificationMessage = NotificationMessageFactory.message(status,
+                        NotificationActionButtonInfo.ButtonAction.LAUNCH_DATE_SETTINGS);
+                break;
+            //Other special handlers will go here when implemented
+            default:
+                if (UpdateHelper.isCombinedErrorMessage(errorMessage)) {
+                    Pair<String, String> resourceAndMessage =
+                            UpdateHelper.splitCombinedErrorMessage(errorMessage);
+                    notificationMessage =
+                            NotificationMessageFactory.message(AppInstallStatus.InvalidResource,
+                                    new String[]{null, resourceAndMessage.first, resourceAndMessage.second},
+                                    NotificationActionButtonInfo.ButtonAction.NONE);
+                } else if (!"".equals(errorMessage)) {
+                    notificationMessage =
+                            NotificationMessageFactory.message(AppInstallStatus.UpdateFailedGeneral,
+                                    new String[]{null, errorMessage, null},
+                                    NotificationActionButtonInfo.ButtonAction.NONE);
+                }
 
-        if (notificationMessage != null) {
-            CommCareApplication.notificationManager()
-                    .reportNotificationMessage(notificationMessage, true);
-            uiController.setNotificationsVisible();
+                break;
         }
+
+        CommCareApplication.notificationManager()
```

Comment:
we should add back `if (notificationMessage != null) {` check here since notification can be null now

Reply:
Agreed!